### PR TITLE
Marks reason correctly for pod failures from preemption

### PR DESCRIPTION
**Pull request by dposada**
## Changes proposed in this PR

Making pods that fail and have a `:pod-preempted-timestamp` go down the `handle-pod-preemption` path so that they get the appropriate `Agent removed` failure reason.

## Why are we making these changes?

Currently, pod failures from node preemption result in the confusing and misleading `Unknown mesos reason`.

### CODE REVIEW

File: scheduler/src/cook/kubernetes/controller.clj [hunk 1]
```diff
@@ -302,6 +302,16 @@
   ; to deleting from the map, into (missing,missing) state.
   {:cook-expected-state :cook-expected-state/completed})
 
+(defn handle-pod-failed
+  "Handles pod failure, checking to see if the failure was caused by pod preemption."
+  [{:keys [name] :as compute-cluster} pod-name {:keys [synthesized-state] :as k8s-actual-state-dict}]
+  (if (some-> synthesized-state :pod-preempted-timestamp)
+    (do
+      (log/info "In compute cluster" name ", pod" pod-name
```
Comment (scrosby) [inline scheduler/src/cook/kubernetes/controller.clj:310]: I"d reword this to be more concrete and clear. Maybe something like:

"has a preemped timestamp and is treated as preempted"

Comment (dposada) [reply scheduler/src/cook/kubernetes/controller.clj]: Done, let me know what you think

Comment (scrosby) [reply scheduler/src/cook/kubernetes/controller.clj]: lgtm
